### PR TITLE
[1.15] daemon: Cleanup after a failed start

### PR DESCRIPTION
**Pull request by antonipp**
Backport https://github.com/cilium/cilium/pull/32673 to 1.15

This required backporting this commit as well: https://github.com/cilium/cilium/commit/f21f303003d9026a925cf5a838ff528ea977f202

Fixes: https://github.com/cilium/cilium/issues/32647

```release-note
cilium-agent: Fix crash due to skipped resource cleanup when agent is stopping due to failed start.
```

### CODE REVIEW

**Comment by joestringer** [comment]
Could you speak a bit about whether you can reproduce the issue you see and how much testing you have performed with this change? The bug being fixed here does not seem to be catastrophic or security-relevant ([criteria](https://docs.cilium.io/en/stable/contributing/release/backports/#backport-criteria-for-x-y-1-z-and-x-y-2-z)), so I'm not sure whether applying this patch to v1.15 is worth the risk. Startup/shutdown ordering issues can be racy and hard to follow, so I'd be concerned about solving one issue then creating another different issue.

**Comment by antonipp** [comment]
Yeah, tbh it's an intermittent issue that we've seen only a handful of times and I haven't managed to reliably reproduce it:
https://github.com/cilium/cilium/issues/32647#issuecomment-2563532891

Since I saw that the original PR had the "backport needed" label, I thought that I might as well do it but I agree with you that it's not a critical bug. So if you think it's not worth it, I am fine with closing the PR.

**Comment by christarazi** [comment]
> Yeah, tbh it's an intermittent issue that we've seen only a handful of times and I haven't managed to reliably reproduce it: [#32647 (comment)](https://github.com/cilium/cilium/issues/32647#issuecomment-2563532891)
> 
> Since I saw that the original PR had the "backport needed" label, I thought that I might as well do it but I agree with you that it's not a critical bug. So if you think it's not worth it, I am fine with closing the PR.

@antonipp Given the [backport criteria](https://docs.cilium.io/en/stable/contributing/release/backports/#backport-criteria-for-x-y-1-z-and-x-y-2-z) and your above statement, it seems like this PR doesn't meet the requirements. If you agree, feel free to close the PR.
